src/main/java/org/springframework/data/couchbase/repository/query/PartTreeN1qlBasedQuery.java

```diff
@@ -50,16 +50,21 @@
 public class PartTreeN1qlBasedQuery extends AbstractN1qlBasedQuery {
 
   private final PartTree partTree;
-  private JsonValue placeHolderValues;
+  private ThreadLocal<JsonValue> placeHolderValues;
 
   public PartTreeN1qlBasedQuery(CouchbaseQueryMethod queryMethod, CouchbaseOperations couchbaseOperations) {
     super(queryMethod, couchbaseOperations);
     this.partTree = new PartTree(queryMethod.getName(), queryMethod.getEntityInformation().getJavaType());
+    this.placeHolderValues = new ThreadLocal<JsonValue>() {
+      @Override public JsonValue initialValue() {
+        return JsonArray.create();
```

Comment:
looks like this was previously uninitialized (the inner value) should this still be null or does this any harm? My assumption is no but still checking

Reply:
This does no harm - IIRC not initializing it was a problem.